GEOS LDFLAGS overwrite created a regression when configuring a build against Android NDK

This commit (https://github.com/OSGeo/gdal/commit/2a19767c76429ee03d5d0d41b55bb38ac7775c6a#diff-3c73726b523598e4e4779ab40bddce9bb1eb99c79244a3c2777b950c951064bd) caused a regression when building gdal in an Android NDK environment. This was discovered when I was updating the gdal recipe on OSGeo4A (QField's Android SDK).

Removing the LDFLAGS overwrite/whipe (as done here https://github.com/opengisch/OSGeo4A/pull/120/files#diff-beb9e4f3ba0c2c87c63fa93cbecc455dc7d2a8ec1fae998268ba01ff0c314915R36 ) fixes the configuration's GEOS handling.

@QuLogic , pinging you on this as you were the author of the above-mentioned commit.

The configuration error I was getting is:

```
checking for geos-config... /usr/src/build/stage/arm64-v8a/bin/geos-config
checking for GEOS version >= 3.1.0... yes
checking for GEOSversion in -lgeos_c... no
configure: error: GEOS library could not be linked
The command '/bin/sh -c ARCHES="${ARCHES}" /usr/src/build_arches.sh' returned a non-zero code: 255

```

--- a/gdal/configure
+++ b/gdal/configure
@@ -38806,7 +38806,7 @@ $as_echo "$as_me: WARNING: GEOS was found on your system, but geos-config report
       ax_save_LIBS="${LIBS}"
       LIBS=${GEOS_LIBS}
       ax_save_CFLAGS="${CFLAGS}"
-      CFLAGS="${GEOS_CFLAGS}"
+      CFLAGS="${GEOS_CFLAGS} ${CFLAGS}"
 
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for GEOSversion in -lgeos_c" >&5
 $as_echo_n "checking for GEOSversion in -lgeos_c... " >&6; }
@@ -39104,7 +39104,7 @@ $as_echo "$as_me: WARNING: SFCGAL was found on your system, but sfcgal-config re
       ax_save_LIBS="${LIBS}"
       LIBS=${SFCGAL_LIBS}
       ax_save_CFLAGS="${CFLAGS}"
-      CFLAGS="${SFCGAL_CFLAGS}"
+      CFLAGS="${SFCGAL_CFLAGS} ${CFLAGS}"
 
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for sfcgal_version in -lSFCGAL" >&5
 $as_echo_n "checking for sfcgal_version in -lSFCGAL... " >&6; }

--- a/gdal/configure
+++ b/gdal/configure
@@ -38807,8 +38807,6 @@ $as_echo "$as_me: WARNING: GEOS was found on your system, but geos-config report
       LIBS=${GEOS_LIBS}
       ax_save_CFLAGS="${CFLAGS}"
       CFLAGS="${GEOS_CFLAGS}"
-      ax_save_LDFLAGS="${LDFLAGS}"
-      LDFLAGS=""
 
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for GEOSversion in -lgeos_c" >&5
 $as_echo_n "checking for GEOSversion in -lgeos_c... " >&6; }
@@ -38868,7 +38866,6 @@ $as_echo "$as_me: WARNING: GEOS was found on your system, but the library could 
 
       CFLAGS="${ax_save_CFLAGS}"
       LIBS="${ax_save_LIBS}"
-      LDFLAGS="${ax_save_LDFLAGS}"
 
     fi
 
@@ -39108,8 +39105,6 @@ $as_echo "$as_me: WARNING: SFCGAL was found on your system, but sfcgal-config re
       LIBS=${SFCGAL_LIBS}
       ax_save_CFLAGS="${CFLAGS}"
       CFLAGS="${SFCGAL_CFLAGS}"
-      ax_save_LDFLAGS="${LDFLAGS}"
-      LDFLAGS=""
 
       { $as_echo "$as_me:${as_lineno-$LINENO}: checking for sfcgal_version in -lSFCGAL" >&5
 $as_echo_n "checking for sfcgal_version in -lSFCGAL... " >&6; }
@@ -39169,7 +39164,6 @@ $as_echo "$as_me: WARNING: SFCGAL was found on your system, but the library coul
 
       CFLAGS="${ax_save_CFLAGS}"
       LIBS="${ax_save_LIBS}"
-      LDFLAGS="${ax_save_LDFLAGS}"
 
     fi
 

--- a/gdal/m4/geos.m4
+++ b/gdal/m4/geos.m4
@@ -131,7 +131,7 @@ AC_DEFUN([GEOS_INIT],[
       ax_save_LIBS="${LIBS}"
       LIBS=${GEOS_LIBS}
       ax_save_CFLAGS="${CFLAGS}"
-      CFLAGS="${GEOS_CFLAGS}"
+      CFLAGS="${GEOS_CFLAGS} ${CFLAGS}"
 
       AC_CHECK_LIB([geos_c],
         [GEOSversion],

--- a/gdal/m4/geos.m4
+++ b/gdal/m4/geos.m4
@@ -132,8 +132,6 @@ AC_DEFUN([GEOS_INIT],[
       LIBS=${GEOS_LIBS}
       ax_save_CFLAGS="${CFLAGS}"
       CFLAGS="${GEOS_CFLAGS}"
-      ax_save_LDFLAGS="${LDFLAGS}"
-      LDFLAGS=""
 
       AC_CHECK_LIB([geos_c],
         [GEOSversion],
@@ -155,7 +153,6 @@ AC_DEFUN([GEOS_INIT],[
 
       CFLAGS="${ax_save_CFLAGS}"
       LIBS="${ax_save_LIBS}"
-      LDFLAGS="${ax_save_LDFLAGS}"
 
     fi
 

--- a/gdal/m4/sfcgal.m4
+++ b/gdal/m4/sfcgal.m4
@@ -131,8 +131,6 @@ AC_DEFUN([SFCGAL_INIT],[
       LIBS=${SFCGAL_LIBS}
       ax_save_CFLAGS="${CFLAGS}"
       CFLAGS="${SFCGAL_CFLAGS}"
-      ax_save_LDFLAGS="${LDFLAGS}"
-      LDFLAGS=""
 
       AC_CHECK_LIB([SFCGAL],
         [sfcgal_version],
@@ -154,7 +152,6 @@ AC_DEFUN([SFCGAL_INIT],[
 
       CFLAGS="${ax_save_CFLAGS}"
       LIBS="${ax_save_LIBS}"
-      LDFLAGS="${ax_save_LDFLAGS}"
 
     fi
 

--- a/gdal/m4/sfcgal.m4
+++ b/gdal/m4/sfcgal.m4
@@ -130,7 +130,7 @@ AC_DEFUN([SFCGAL_INIT],[
       ax_save_LIBS="${LIBS}"
       LIBS=${SFCGAL_LIBS}
       ax_save_CFLAGS="${CFLAGS}"
-      CFLAGS="${SFCGAL_CFLAGS}"
+      CFLAGS="${SFCGAL_CFLAGS} ${CFLAGS}"
 
       AC_CHECK_LIB([SFCGAL],
         [sfcgal_version],

